src/main/scala/com/microsoft/hyperspace/actions/RefreshAction.scala

```diff
@@ -42,12 +42,19 @@ class RefreshAction(
 
   private lazy val previousIndexLogEntry = previousLogEntry.asInstanceOf[IndexLogEntry]
 
-  // Deserialize the plan and create a df.
+  // Reconstruct a df from schema
   private lazy val df = {
-    val serializedPlan = previousIndexLogEntry.source.plan.properties.rawPlan
-    val plan = LogicalPlanSerDeUtils.deserialize(serializedPlan, spark)
-    val qe = spark.sessionState.executePlan(plan)
-    new Dataset[Row](spark, plan, RowEncoder(qe.analyzed.schema))
+    val rels = previousIndexLogEntry.relations
+    // Currently we only support to create an index on a LogicalRelation
+    assert(rels.size == 1)
+    val dataSchema = DataType.fromJson(rels.head.dataSchemaJson).asInstanceOf[StructType]
+    // "path" key in options incurs to read data twice unexpectedly
```

Comment:
nit: can we end a sentence with "."? Btw, did you meaning listing files (not reading data)`// Remove "path" key in options to prevent listing files twice.`?

Reply:
Oh I see. The data is actually read twice.

Reply:
Ah, I see what's happening and I think this could be a bug in Spark. I have a potential fix and will follow up on this.

```
// Both are same and will read the data twice.
spark.read.option("path", "/tmp/parquet_data/").parquet("/tmp/parquet_data/").show
spark.read.parquet("/tmp/parquet_data/", "/tmp/parquet_data/").show
// But the following is fine:
spark.read.option("path", "/tmp/parquet_data/").format("parquet").load("/tmp/parquet_data/").show
```

Putting the above behavior aside, if the user created a `df` with:
```
// Intention is to read three times.
val df = spark.read.option("path", "/tmp/parquet_data").format("parquet").load("/tmp/parquet_data/", "/tmp/parquet_data/")
```
If the user intention was reading the data three times, we can't remove "path" in option?

Reply:
I think this could be a bug. In our refresh-full test, only the index of csv format has the "path" key in its options even I read it without "path" option.

Reply:
And it seems `rootPaths` are stored correctly if the user put multiple paths at load:
```
val df = spark.read.option("path", "/tmp/parquet_data").format("parquet").load("/tmp/parquet_data/", "/tmp/parquet_data/")
```
In this case `rootPaths` might be ["/tmp/parquet_data", "/tmp/parquet_data/", "tmp/parquet_data"]

So I think excluding "path" key here would be okay.

Reply:
There is an inconsistent behavior (Spark 2.4.6):

```
scala> val df = spark.read.option("path","table2").load("table2","table2")
df: org.apache.spark.sql.DataFrame = [id: int, name: string ... 1 more field]

scala> df.show
+---+-----+---+
| id| name|age|
+---+-----+---+
|  1|name1|  4|
|  2|name2|  3|
|  1|name1|  4|
|  2|name2|  3|
|  1|name1|  4|
|  2|name2|  3|
+---+-----+---+


scala> val df = spark.read.option("path","table2").load("table2")
df: org.apache.spark.sql.DataFrame = [id: int, name: string ... 1 more field]

scala> df.show
+---+-----+---+
| id| name|age|
+---+-----+---+
|  1|name1|  4|
|  2|name2|  3|
+---+-----+---+
```

Reply:
> There is an inconsistent behavior (Spark 2.4.6/3.0.0):

Yup, that's what I meant [here](https://github.com/microsoft/hyperspace/pull/99#discussion_r464039124), which I am creating a PR against OSS Spark.

Here are the follow up action items:
1. Should we move removing `path` to the creation side so that it's not in the log entry? The `path` "can" be added by [Spark](https://github.com/apache/spark/blob/71aea02e9ffb0c6f7c72c91054c2a4653e22e801/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L232) (not by user), so it is not useful for debugability anyway. Let me know if there is a use case for storing `path` in the log entry.
2. Can we add a test that tests against all the scenarios (useful for reference as well)?
```
spark.read.format("parquet").load("f1")
spark.read.format("parquet").load("f1", "f2")
spark.read.format("parquet").option("path", "blah").load("f1")
spark.read.format("parquet").option("path", "blah").load("f1", "f2")
spark.read.parquet("f1")
spark.read.parquet("f1", "f2")
spark.read.option("path", "blah").parquet("f1")
spark.read.option("path", "blah").parquet("f1", "f2")
```